azurelinuxagent/common/protocol/extensions_goal_state_from_vm_settings.py

```diff
@@ -267,10 +269,16 @@ def _parse_agent_manifests(self, vm_settings):
         for family in families:
             name = family["name"]
             version = family.get("version")
+            is_version_from_rsm = family.get("isVersionFromRSM")
+            is_vm_enabled_for_rsm_upgrades = family.get("isVMEnabledForRSMUpgrades")
             uris = family.get("uris")
             if uris is None:
                 uris = []
             agent_family = VMAgentFamily(name, version)
+            if is_version_from_rsm is not None:
```

Comment:
This is confusing. Why are you enforcing passing a version to VMAgentFamily, but not passing is_version_from_rsm and is_vm_enabled_for_rsm_updates? in other parts of the code you are checking for version to not be None, so apparently all 3 of them can be None